python_modules/dagster/dagster/_core/execution/api.py

```diff
@@ -1096,6 +1096,9 @@ def pipeline_execution_iterator(
             elif event.is_resource_init_failure and event.step_key:
                 failed_steps.append(event.step_key)
 
+            # Telemetry
+            log_step_event(event, pipeline_context)
```

Comment:
`pipeline_execution_iterator` can execute in another process / potentially another pod, so I think this event is going to get lost when not using a local executor.